ui/app/adapters/kv/data.js

```diff
@@ -31,6 +31,16 @@ export default class KvDataAdapter extends ApplicationAdapter {
     });
   }
 
+  fetchSubkeys(query) {
```

Comment:
I put this in `kv/data` (instead of `metadata`) adapter because this endpoint error handles very similar to the `READ` secret data endpoint. For example, when a version has been deleted there is still metadata returned by the endpoint even if there are no `subkeys` returned.

When we implement this request, we can reuse the error handling logic below in this file. 

https://github.com/hashicorp/vault/blob/02950f4b02a298424ce6bba0b4154bc5e05161ca/ui/app/adapters/kv/data.js#L89-L102

Reply:
One of the reasons Chelsea and I opted to put all `data` related endpoints and `metadata` into their own adapter is to map endpoints by their base names. So when i look at the data adapter I expect all endpoints to have data in the base url and the same for metadata. It helped us a lot in development but we thought it might also help future debugging efforts.

Any considerations to making kv/subkeys adapter to follow suite? I understand if you think it's overkill, but want to point out the current separation pattern.

Reply:
...okay, I'm remember delete functionality now. That was the exception, so maybe subkeys is an addition to that exception—just thinking out loud.

Reply:
Great question! I didn't want to create a `kv/subkeys` model because it's just this one request and there's no fancy schema that we'd need a model for. Plus the error handling is going to be the exact same as `kv/data` so I can share the logic above between both `queryRecord` here and in the `fetchSubkeys` request. I'm just waiting to add that logic when actually developing the feature. 

So, even though `data` isn't in the request, it behaves similarly to READ secret data that putting here seemed to make the most sense. What do you think?

Reply:
Works for me! Thanks for the explanation.